src/dehtml.rs

```diff
@@ -172,15 +230,57 @@ fn dehtml_starttag_cb<B: std::io::BufRead>(
             }
         }
         "b" | "strong" => {
-            dehtml.strbuilder += "*";
+            if dehtml.get_add_text() != AddText::No {
+                dehtml.strbuilder += "*";
+            }
         }
         "i" | "em" => {
-            dehtml.strbuilder += "_";
+            if dehtml.get_add_text() != AddText::No {
+                dehtml.strbuilder += "_";
+            }
         }
         _ => {}
     }
 }
 
+/// In order to know when a specific tag is closed, we need to count the opening and closing tags.
+/// The `counts`s are stored in the `Dehtml` struct.
+fn pop_tag(count: &mut Option<u32>) {
+    if let Some(ref mut divs) = count {
+        if *divs > 1 {
```

Comment:
Now why not use a simple `u32` if `Some(0)` is never used?

Reply:
because someday ago I heard that in Rust you shouldn't use special values but use Option, I don't know when and where :) I don't actually have an opinion on this, code should be about equally long

Reply:
`0` is a fine special value for a counter, this applies more to things like using `-1` to indicate an error